src/OAuth2.php

```diff
@@ -1376,23 +1378,60 @@ private function coerceUri($uri)
 
     /**
      * @param string $idToken
-     * @param string|array|null $publicKey
-     * @param array $allowedAlgs
+     * @param Key|Key[]|string|array|null $publicKey
+     * @param string|array $allowedAlg
      * @return object
      */
     private function jwtDecode($idToken, $publicKey, $allowedAlgs)
     {
-        return JWT::decode($idToken, $publicKey, $allowedAlgs);
+        $keys = $this->getFirebaseJwtKeys($publicKey, $allowedAlgs);
+
+        // Default exception if none are caught
```

Comment:
```suggestion
        // Default exception if all are caught
```

Reply:
Actually I think what i have here is better. The reason is that the API exception being thrown is the default to preserve backwards compatibility. The validation doesn't happen here, but we pretend like it does for BC.